jack.py

```diff
@@ -739,12 +739,17 @@ def set_process_callback(self, callback, userdata=None):
             is called.
 
         """
-        @self._callback("JackProcessCallback", error=STOP_CALLING)
-        def callback_wrapper(frames, _):
-            return callback(frames, userdata)
+        if isinstance(callback, _ffi.CData):
+            callback_wrapper = callback
+            arg = _ffi.cast("void *", userdata)
```

Comment:
I'm not quite sure, but I suspect the cast to `void*` isn't even necessary since CFFI seems to do the cast implicitly.